ARCHITECTURE.md

```diff
@@ -0,0 +1,226 @@
+# Istio operator architecture
+
+## Introduction
+
+The operator code is divided roughly into five areas:
+
+1. [New API](The_new_API) and related infrastructure. The new API is expressed as a
+[proto](https://github.com/istio/operator/blob/master/pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto) and
+compiled to [Go structs](https://github.com/istio/operator/blob/master/pkg/apis/istio/v1alpha2/istiocontrolplane_types.pb.go). The new
+API has pass-through fields to the Helm values.yaml API, but these are additionally validated through a
+[schema](https://github.com/istio/operator/blob/master/pkg/apis/istio/v1alpha2/values_types.go).
+1. [Controller](K8s_controller) code. The code comprises the K8s listener, webhook and logic for reconciling the cluster
+to an IstioControlPlane CR.
+1. [Manifest creation](Manifest_creation) code. Manifest rendering is done by overlaying user settings on top of a
+selected profile and passing the resulting values to a renderer in the Helm library to create manifests. Further
+overlays on the created manifests may be applied if they are defined in the user settings.
+1. [CLI](CLI) code. CLI code shares the IstioControlPlane API with the controller, but allows manifests to be generated
```

Comment:
We also probably want an operational status/check area. This would be used by the operator for self-healing and CRD status. It could also be exposed through the CLI, e.g. `istioctl check`.